foundation/foundation-messaging/foundation-messaging-python/aissemble-foundation-messaging-python-client/src/aissemble_messaging/messaging_client.py

```diff
@@ -167,9 +169,29 @@ def start_service_jvm(self):
         path_to_jar = str(RESDIR.joinpath("classpath/*"))
         classpath = path_to_jar
 
+        py4j_jar_path = self.get_py4j_jar_path()
         return launch_gateway(
+            jarpath=py4j_jar_path,
             classpath=classpath,
             die_on_exit=True,
             redirect_stdout=sys.__stdout__,
             redirect_stderr=sys.__stderr__,
         )
+
+    def get_py4j_jar_path(self):
+        """
+        Tries to find the path where the py4j jar is located.
+        """
+        jar_path = find_jar_path()
+        if not jar_path:
+            jar_file = "py4j{0}.jar".format(__version__)
+            ## if jar_path not found by py4j, the `sys.prefix` could be changed
+            ## set the py4j jar path to default /usr/local/share directory where
+            ## the local installations are
+            default_py4j_jar_path = "/usr/local/share/py4j/" + jar_file
+            if os.path.exists(default_py4j_jar_path):
+                self.logger.info("Setting py4j jar path to default path")
```

Comment:
S; Might be more helpful here to log the actual path, as default path isn't really clear what we're setting it to:
`"Setting py4j jar path to: " + default_py4j_jar_path`